app/bundles/EmailBundle/Helper/MessageHelper.php

```diff
@@ -2144,4 +2146,67 @@ public static function parseAddressList($addresses)
 
         return $results;
     }
+    
+    public function processReplyMail($mailId, $refid)
+    {
+        $this->logger->debug('Processing reply mail.');
+        $imapHelper     = $this->factory->get('mautic.helper.mailbox');
```

Comment:
Avoiding the MauticFactory usage in this class is harder, but to make the future refactoring easier please define the helpers in constructor as class properties and use those instead of getting them from the factory in the methods.

Reply:
Done